client/src/graphs/templates/ui/TemplateItem.vue

```diff
@@ -0,0 +1,54 @@
+<script setup lang="ts">
+import GButton from "@ui/graph/button/GButton.vue";
+import type { GraphTemplate } from "../types";
+import { ref, toRefs } from "vue";
+import colors from "@utils/colors";
+
+const props = defineProps<{
+  template: GraphTemplate;
+  loadTemplate: (id: string) => void;
+  deleteTemplate: (id: string) => void;
+}>();
+
+const { template } = toRefs(props);
+
+const hovered = ref(false);
+</script>
+
+<template>
+  <div
+    @mouseenter="hovered = true"
+    @mouseleave="hovered = false"
+    class="relative flex flex-col"
+  >
+    <div
+      class="transition duration-200 p-2"
+      :style="{
+        opacity: hovered ? 0.5 : 1,
+      }"
+    >
+      <h3 class="text-xl">{{ template.title }}</h3>
+      <p class="text-sm font-normal">{{ template.description }}</p>
+      <img
+        :src="template.thumbnail"
+        class="aspect-video h-52 object-contain p-2"
+      />
+    </div>
+    <div
+      class="absolute w-full h-full z-10 flex items-center justify-center transition duration-200"
+      :style="{
+        opacity: hovered ? 1 : 0,
+      }"
+    >
+      <div class="flex gap-2">
+        <GButton @click="loadTemplate(template.id)">Load</GButton>
+        <GButton
+          v-if="!template.productId"
```

Comment:
this is meant to be if its user added? If so thats confusing as fuck for a non-informed individual. so make a fucking name in the template that is "isUserAdded"

Reply:
or make a more sustainable solution then using productId cus what if the user wants to add a template that is specific to a product